### ISSUE TITLE
Regression with enums in 1.1.174

### ISSUE THREAD

**Issue by tasn** [microsoft/pyright#2367]
**Describe the bug**
Getting `error: Base class "BaseIntEnum" is marked final and cannot be subclassed` although subclassing enums is allowed in some circumstances (like the one we have).

See https://docs.python.org/3/library/enum.html#restricted-enum-subclassing

**To Reproduce**
```
# foo.py:
from enum import Enum


class BaseStringEnum(str, Enum):
    @classmethod
    def foobar(cls):
        return "foobar"


class FooEnum(BaseStringEnum):
    A = "A"
    B = "B"
```
run:
```
$ npx pyright@1.1.174 foo.py
error: Base class "BaseStringEnum" is marked final and cannot be subclassed
```

**Expected behavior**
This should not be an error.

**Screenshots or Code**
See above.

**VS Code extension or command-line**
Command line 1.1.174

**Additional context**
Broke in 1.1.174, works in 1.1.173

**Comment by erictraut** [comment]
Thanks for reporting, and apologies for the regression. I wasn't aware that this was allowed.

**Comment by tasn** [comment]
No worries, thanks for the prompt response. We've just pinned 1.1.173 for now to avoid it, so no major disruption on our end. :)

**Comment by erictraut** [comment]
I've backed out the change that caused the regression. I'll need to rethink how to implement this check and take another run at it. In the meantime, the regression will be fixed, and I've added a test case based on your sample above to prevent future regressions in this area.

**Comment by tasn** [comment]
Please double-check my understanding of the Python docs, but I think it's correct. Right?

**Comment by erictraut** [comment]
Yes, I agree with your interpretation, and I've verified the runtime behavior.

**Comment by sobolevn** [comment]
Just to be sure: is this the same case we are testing [here](https://github.com/python/mypy/pull/11247/files#diff-ebc735ad8fa6eb42d5390fa4fa9ef60581764c7bbfb5eafcde16f5e40c330c84R1424-R1446)?

**Comment by erictraut** [comment]
Yes, looks like you've got this case covered in your test cases.

**Comment by bellini666** [comment]
Hey guys,

Just came here to open an issue and found this. The same happened to a project of mine where we use `TextChoices`, which is a subclass of `enum.Enum`, and they all are giving me the same errors. Here is a simple test in case you need it:

```python
from django.db import models

class FooChoices(models.TextChoices):
    BAR = "bar", "Bar"
    BIN = "bin", "Bin"
```

Run `pip install django django-types` in the venv and then pyright, it will give the same issues.

**Comment by erictraut** [comment]
I've published pyright 1.1.175 as a "hot fix" release that addresses this regression. Sorry for any inconvenience.
